packages/cadl-ranch-specs/http/payload/multipart/main.tsp

```diff
@@ -0,0 +1,41 @@
+import "@typespec/http";
+import "@azure-tools/cadl-ranch-expect";
+
+using TypeSpec.Http;
+
+@doc("Test for multipart")
+@scenarioService("/multipart")
+namespace Payload.MultiPart;
+
+model MultiPartRequest {
+  id: string;
+  profileImage: bytes;
```

Comment:
we should add encoding for `bytes`, such as `binary` for the file content. Or we may need to make a presuppose that in multipart, the default encoding of `bytes` is `binary`, others content-type, the default is `base64`. May need typespec design for this.

Reply:
Do we have `binary`?
https://github.com/microsoft/typespec/blob/main/packages/compiler/lib/decorators.tsp#L424-L437

base64 is only for application/json

Reply:
No, current no. So we may need typespec to define this.

Reply:
I think for `bytes`, we delegate the encoding to caller to decide. If it's a binary file, then no encoding is required.

Reply:
I am not sure what you mean you delegate to the caller to decide, the spec is what specify how the data is sent and there shouldn't be room for interpretation by the caller if they should send it encoded a certain way or not.

1. `bytes` in the body of `application/octet-stream` is the raw binary payload/file
2. `bytes` inside of a json object should be encoded as base64 by default
3. `bytes` inside a multipart part (for now) is assumed to be the raw bytes
4. `bytes` as the body of `application/json` right now this respect the `base64` encoding but we are discussing if this shouldn't change as it cause issues.

Reply:
Reminder, this playgrouynd has been discussed and clarifies the questions here:
https://cadlplayground.z22.web.core.windows.net/cadl-azure/?c=aW1wb3J0ICJAdHlwZXNwZWMvaHR0cCI7CnVzaW5nIFR5cGVTcGVjLkh0dHA7CgpvcCB1cGxvYWRJbWFnZShAaGVhZGVyIGNvbnRlbnTELjogIm11bHRpcGFydC9mb3JtLWRhdGEiLCBAYm9keSBpxDg6IE3EI1BhcnRSZXF1ZXN0KTogdm9pZDsKCm1vZGVsIEFkZHJlc3MgewogIHN0cmVldDrECGluZzsKICBjaXR5yhB9CgpAZW5jb2RlKCJiaW5hcnkiKSAgLy8gU2VlIOQAznM6Ly9naXRodWIuY29tL21pY3Jvc29mdC%2FpAPRpc3N1ZXMvMjM3MApzY2FsYXIgxkogZXh0ZW5kcyBieXRlc%2BkAn%2FAAv%2BUAqGlk6QCUxH5GaWVsZCAiaWQiIGlzIHRleHQvcGxhaW4KICBh5gDeOugA58wwxxzFNWFuIGFwcGxpY2F0aW9uL2pzb24KCiAgcHJvZmlsZeUBgzrnAKk75QDyx0TMIcZJy0ZvY3RldC3kAVVhbcZOZXZpb3VzQcV4ZeoAlltd5QCYVGhpc8RET05FIOQB1SwgYXMgYSBKU09OIGFycmF5xUVpY3R1csQ95gCPzzxhcyBtYW55xUDEDW5lY2Vzc2FyeQogIOQBwAovKioKClBPU1QgL%2BYCYCBIVFRQLzEuMQpD5gJcLUxlbmd0aDogNDI4yRTmAnHzAnA7IGJvdW5kYXJ5PWFiY2RlMTIzNDUKCi0tyw7IRURpc3Bvc2nkAR06IMtCbmFtZT3kAcDPb%2BsBywoxMjNlNDU2Ny1lODliLTEyZDMtYTQ1Ni00MjY2NTU0NDAwMDDfdtV26AIGz3vyAg7kAnki5gMiIjogIjMsIEdhcmRlbiBTdCIsxBzkAyzEGkhpbGxzYmVyeSwgVVQiCn3%2FAJb1AJbsAlggIjsgxA3GGuUD1zEucG5n%2FACz7gJ7e%2BKApsRC6AQ%2B4oCm7gCWLS3%2FAJjqAJjuAsP%2FATduCgpb7wE4NDIsIEhpdGNoaGlrZXJz8QE%2BTmV3LVlvcmssIE5ZIgp9LM8%2FMjIsIEFjYWNpYSBBds46TG9uZG9uLCBVSyIKfV3%2FAXb2AN7nA1ztAXHHFf8Bc%2F8Bc%2FIBc%2F8Ak%2F8Ak8UVMv8Ak%2F8Ak%2FIAkwogKi8%3D&e=%40typespec%2Fopenapi3&options=%7B%7D

`profileImage` here is bytes `application/octet-stream`, there is no ambiguity

Reply:
> I am not sure what you mean you delegate to the caller to decide, the spec is what specify how the data is sent and there shouldn't be room for interpretation by the caller if they should send it encoded a certain way or not.
> 
> 1. `bytes` in the body of `application/octet-stream` is the raw binary payload/file
> 2. `bytes` inside of a json object should be encoded as base64 by default
> 3. `bytes` inside a multipart part (for now) is assumed to be the raw bytes
> 4. `bytes` as the body of `application/json` right now this respect the `base64` encoding but we are discussing if this shouldn't change as it cause issues.

@timotheeguerin I mean 3rd case, since we're discussing test cases for `multipart/form-data`.